File: analysis/import/SonarImporter/src/main/java/de/maibornwolff/codecharta/importer/sonar/model/ComponentMap.java
package de.maibornwolff.codecharta.importer.sonar.model;

import java.util.HashMap;
import java.util.Map;
import java.util.stream.Stream;

public class ComponentMap {
    private final Map<String, Component> components = new HashMap<>();

    public void updateComponent(Component component) {
        if (components.containsKey(component.getKey())) {
            Component updateableComponent = components.get(component.getKey());
            updateableComponent.getMeasures().addAll(component.getMeasures());
        } else {
            components.put(component.getKey(), component);
        }
    }

    public Stream<Component> getComponentStream() {
        return components.values().stream();
    }

}

File: analysis/import/SonarImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/sonar/model/ComponentMap.kt
package de.maibornwolff.codecharta.importer.sonar.model

class ComponentMap {
    private val components = mutableMapOf<String, Component>()

    val componentList: MutableCollection<Component>
        get() = components.values

    fun updateComponent(component: Component) {
        if (components.containsKey(component.key)) {
            val updateableComponent = components[component.key]!!
            updateableComponent.measures!!.addAll(component.measures!!)
        } else {
            components[component.key!!] = component
        }
    }
}
